### ISSUE TITLE
iconv.dll was not found - error running grass.core.create_location

### ISSUE THREAD

**Issue by everydayduffy** [tgrippa/Opensource_OBIA_processing_chain#2]
Hello,

When I try to run:

`grass.core.create_location(user["gisdb"], user["location"], epsg=user["locationepsg"], overwrite=False)`

I get an error. As you can see from the attached image, my database folder and location have been created successfully. 

![error](https://cloud.githubusercontent.com/assets/15983662/25753389/2d17573c-31b3-11e7-8727-1842b8dc222c.PNG)

The traceback reads as follows:

```
---------------------------------------------------------------------------
CalledModuleError                         Traceback (most recent call last)
<ipython-input-12-7d7a61234f98> in <module>()
----> 1 grass.core.create_location(user["gisdb"], user["location"], epsg=user["locationepsg"], overwrite=False)

C:\OSGeo4W64\apps\grass\grass-7.3.svn\etc\python\grass\script\core.pyc in create_location(dbase, location, epsg, proj4, filename, wkt, datum, datum_trans, desc, overwrite)
   1457     if epsg or proj4 or filename or wkt:
   1458         # FIXME: changing GISDBASE mid-session is not background-job safe
-> 1459         gisdbase = gisenv()['GISDBASE']
   1460         run_command('g.gisenv', set='GISDBASE=%s' % dbase)
   1461     # create dbase if not exists

C:\OSGeo4W64\apps\grass\grass-7.3.svn\etc\python\grass\script\core.pyc in gisenv()
   1008     :return: list of GRASS variables
   1009     """
-> 1010     s = read_command("g.gisenv", flags='n')
   1011     return parse_key_val(s)
   1012 

C:\OSGeo4W64\apps\grass\grass-7.3.svn\etc\python\grass\script\core.pyc in read_command(*args, **kwargs)
    472     if _capture_stderr and returncode:
    473         sys.stderr.write(stderr)
--> 474     return handle_errors(returncode, stdout, args, kwargs)
    475 
    476 

C:\OSGeo4W64\apps\grass\grass-7.3.svn\etc\python\grass\script\core.pyc in handle_errors(returncode, result, args, kwargs)
    328         args = make_command(*args, **kwargs)
    329         raise CalledModuleError(module=None, code=repr(args),
--> 330                                 returncode=returncode)
    331 
    332 def start_command(prog, flags=b"", overwrite=False, quiet=False,

CalledModuleError: Module run None ['g.gisenv', '-n'] ended with error
Process ended with non-zero return code -1073741515. See errors in the (error) output.
```

Do you know why this might be please?

**Comment by tgrippa** [comment]
I found a "iconv.dll" file in the "C:\OSGeo4W64\bin" folder. You could try to add it in the PATH, like that:
`os.environ['PATH'] = 'C:\\OSGeo4W64\\bin' + os.pathsep + os.environ['PATH']`

**Comment by everydayduffy** [comment]
Thank you. Adding that to the PATH has removed that issue, but I still have a problem running that section of code:

![error](https://cloud.githubusercontent.com/assets/15983662/25797456/36a33964-33d5-11e7-841d-1e7b446c364a.PNG)

**Comment by tgrippa** [comment]
Try this : 
1) Add the following line at the top of your notebook (at the same place you set up your environmental variables) : `grass7bin_win = 'C:\\OSGeo4W64\\bin\\grass73svn.bat'`
2) Delete the location manually (in the file explorer)
3) Replace the content of cell for automatic creation of GRASS location by this : [https://gist.github.com/tgrippa/8d77d9083a9654085b1089ad79cd22a6](https://gist.github.com/tgrippa/8d77d9083a9654085b1089ad79cd22a6)

If it works I will make the modification available for next users.

**Comment by everydayduffy** [comment]
Hello,

1. Done.
2. Do you mean delete `C:\\OSGeo4W64\\bin\\grass73svn.bat` ?
3. Do you mean replace `grass.core.create_location(user["gisdb"], user["location"], epsg=user["locationepsg"], overwrite=False)` with the code in your gist? 

Thanks!

**Comment by tgrippa** [comment]
2. No, **dont** delete the grass73svn.bat file. What I would say is to delete the GRASS location which exists at the moment (in your GRASSDATA folder). Open the file explorer and go to your GRASSDATA folder, then remove everything inside (I guess you didn't have any important data in your GRASSDATA).
3. You can just remove the content of the cell beginning with the '## Automatic creation of GRASS location is doesn't exist'

**Comment by everydayduffy** [comment]
Hello,

Your suggestions have worked. Solved. Thank you.

**Comment by tgrippa** [comment]
It has been fixed in the notebook (commit 8d938da835c92a3f321190cf628726c5e5f86ef2).
